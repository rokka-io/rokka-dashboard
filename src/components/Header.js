import PropTypes from 'prop-types'
import React from 'react'
import { Link } from 'react-router-dom'
import { toggleSidebar, logout } from '../state'
import avatarIcon from '../img/avatar-placeholder.svg'
import logoutIcon from '../img/logout-icon.svg'
import cx from 'classnames'

const Header = (props) =>
  <header className="rka-header">
    <a href="#" className={cx('rka-header-menu-icon', {'is-active': props.active})}
      onClick={(e) => { toggleSidebar(); e.preventDefault() }} />
    <Link to="/" className="rka-header-logo" />
    <nav className="rka-header-nav">
      <a className="rka-user">
        {/* !user (avatar placeholder) */}
        <div className="rka-avatar">
          <svg className="rka-avatar-icon">
            <use xlinkHref={avatarIcon + '#avatar'} />
          </svg>
        </div>
        <span className="rka-username">{props.auth.organization}</span>
      </a>
      <a href="#" className="rka-logout" onClick={(e) => { e.preventDefault(); logout() }}>
        <svg className="rka-logout-icon">
          <use xlinkHref={logoutIcon + '#logout'} />
        </svg>
      </a>
    </nav>
  </header>

Header.propTypes = {
  auth: PropTypes.shape({
    organization: PropTypes.string.isRequired
  }).isRequired,
  active: PropTypes.bool.isRequired
}

export default Header
