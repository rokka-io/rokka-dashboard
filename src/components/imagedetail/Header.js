import PropTypes from 'prop-types'
import React from 'react'
import cx from 'classnames'
import moment from 'moment'
import focusIcon from '../../img/focus-icon.svg'
import linkicon from '../../img/link-icon.svg'

const Header = ({ image, focusMenuActive, onClickToggleFocusMenu, onClickSave }) => (
  <div className="rka-content-header clearfix">
    <div className="flo-l">
      <h1 className="rka-h1">{image.name}</h1>
      <p className="txt-gray-darker mt-sm txt-sm txt-italic">
        <span className="mr-sm">{moment(image.created).fromNow()},</span>
        <span className="mr-sm">{`${Math.round(image.size / 1024)}KB`},</span>
        <span className="mr-sm">{[image.width || '?', image.height || '?'].join('×')},</span>
        <span className="mr-sm">{image.format.toUpperCase()},</span>
        <span className="">
          <a
            className="rka-link rka-hash-link"
            href={image.url}
            target="_blank"
            rel="noopener noreferrer"
          >
            {image.hash}
            <svg className="rka-link-icon">
              <use xlinkHref={linkicon + '#linkicon'} />
            </svg>
          </a>
        </span>
        <span style={{ marginLeft: 30 }}>
          <a
            className="rka-link rka-hash-link"
            href={image.url.replace(image.hash, image.short_hash)}
            target="_blank"
            rel="noopener noreferrer"
          >
            {image.short_hash}
            <svg className="rka-link-icon">
              <use xlinkHref={linkicon + '#linkicon'} />
            </svg>
          </a>
        </span>
      </p>
    </div>
    <div className="flo-r">
      <button
        className={cx('rka-focus-trigger', 'mr-md', { 'is-active': focusMenuActive })}
        onClick={onClickToggleFocusMenu}
      >
        {focusMenuActive ? (
          <span className="rka-focus-icon txt-c txt-lg">×</span>
        ) : (
          <svg className="rka-focus-icon">
            <use xlinkHref={focusIcon + '#focusicon'} />
          </svg>
        )}
      </button>
      <button className="rka-button rka-button-brand mt-sm" onClick={onClickSave}>
        Save
      </button>
    </div>
  </div>
)
Header.propTypes = {
  image: PropTypes.shape({
    name: PropTypes.string.isRequired,
    created: PropTypes.string.isRequired,
    size: PropTypes.number.isRequired,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
    format: PropTypes.string.isRequired,
    hash: PropTypes.string.isRequired
  }).isRequired,
  focusMenuActive: PropTypes.bool.isRequired,
  onClickToggleFocusMenu: PropTypes.func.isRequired,
  onClickSave: PropTypes.func.isRequired
}

export default Header
