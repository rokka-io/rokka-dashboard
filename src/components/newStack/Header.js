import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import Spinner from '../Spinner'

const Header = ({
  updatePreview,
  previewImage = null,
  createStackBtnDisabled = false,
  isPreviewCurrent = false,
  showLoader = false
}) => (
  <div className="bg-white pa-md clearfix">
    <h1 className="rka-h1 flo-l mt-xs">New stack</h1>
    <div className="flo-r">
      {previewImage && (
        <button
          type="button"
          onClick={e => {
            e.preventDefault()
            updatePreview(previewImage)
          }}
          disabled={isPreviewCurrent}
          className="rka-button rka-button-secondary mr-md"
        >
          Update preview
        </button>
      )}
      <button
        className={cx('rka-button rka-button-brand', { 'disabled flo-r': createStackBtnDisabled })}
        type="submit"
      >
        {showLoader ? (
          <div className="sk-cube-small sk-cube-white">
            <Spinner />
          </div>
        ) : (
          'Create stack'
        )}
      </button>
    </div>
  </div>
)
Header.propTypes = {
  updatePreview: PropTypes.func.isRequired,
  previewImage: PropTypes.object,
  createStackBtnDisabled: PropTypes.bool,
  isPreviewCurrent: PropTypes.bool,
  showLoader: PropTypes.bool
}

export default Header
