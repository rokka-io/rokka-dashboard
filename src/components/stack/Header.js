import React from 'react'
import PropTypes from 'prop-types'

const Header = ({ children, title }) => (
  <div className="bg-white pa-md clearfix">
    <h1 className="rka-h1 flo-l mt-xs">{title}</h1>
    <div className="flo-r">{children}</div>
  </div>
)
Header.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired
}

export default Header
