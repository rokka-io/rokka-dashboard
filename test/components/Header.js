import React from 'react'
import Header from '../../src/components/Header'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

test('Header does render', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Header auth={{organization: 'test-organization'}} active={false} />
    </MemoryRouter>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Header does render in active mode', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Header auth={{organization: 'test-organization'}} active />
    </MemoryRouter>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
