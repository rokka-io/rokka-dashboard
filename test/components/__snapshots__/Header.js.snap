// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header does render 1`] = `
<header
  className="rka-header"
>
  <a
    className="rka-header-menu-icon"
    href="#"
    onClick={[Function]}
  />
  <a
    className="rka-header-logo"
    onClick={[Function]}
    style={Object {}}
  />
  <nav
    className="rka-header-nav"
  >
    <a
      className="rka-user"
    >
      <div
        className="rka-avatar"
      >
        <svg
          className="rka-avatar-icon"
        >
          <use
            xlinkHref="avatar-placeholder.svg#avatar"
          />
        </svg>
      </div>
      <span
        className="rka-username"
      >
        test-organization
      </span>
    </a>
    <a
      className="rka-logout"
      href="#"
      onClick={[Function]}
    >
      <svg
        className="rka-logout-icon"
      >
        <use
          xlinkHref="logout-icon.svg#logout"
        />
      </svg>
    </a>
  </nav>
</header>
`;

exports[`Header does render in active mode 1`] = `
<header
  className="rka-header"
>
  <a
    className="rka-header-menu-icon is-active"
    href="#"
    onClick={[Function]}
  />
  <a
    className="rka-header-logo"
    onClick={[Function]}
    style={Object {}}
  />
  <nav
    className="rka-header-nav"
  >
    <a
      className="rka-user"
    >
      <div
        className="rka-avatar"
      >
        <svg
          className="rka-avatar-icon"
        >
          <use
            xlinkHref="avatar-placeholder.svg#avatar"
          />
        </svg>
      </div>
      <span
        className="rka-username"
      >
        test-organization
      </span>
    </a>
    <a
      className="rka-logout"
      href="#"
      onClick={[Function]}
    >
      <svg
        className="rka-logout-icon"
      >
        <use
          xlinkHref="logout-icon.svg#logout"
        />
      </svg>
    </a>
  </nav>
</header>
`;
